### ISSUE TITLE
PPU LLVM Precomp not working properly if MGS4 is updated to 2.00

### ISSUE THREAD

**Issue by ItsLaguna** [RPCS3/rpcs3#12320]
# Quick summary

PPU LLVM Precompilation doesn't work correctly if the game is updated to ver 2.00

## Details

PPU LLVM Precompilation works correctly in the disc version (v1.00), this means that after going through the whole process you won't compile in loading screens, but if it's updated to v2.00 and precompiled, you'll have to compile modules in each loading screen. Can't be reproduced in 25th Anniversary version or Digital version (as both already come as 2.00 and they can't be updated).

Before finding out what it seems to be an issue exclusive to disc versions (which contain the 1.00 data and can be updated) I tried older versions of the emulator in case it was a regression to no avail (will attach some of the logs).

#### Please provide your system configuration:
* OS: Windows 10 Pro
* CPU: Ryzen 7 5800X
* GPU: Nvidia RTX 2060
* Driver version: 516.40.0.0
* RPCS3 version: happens on any version.

#### Please include.
* Anything else you deem to be important

Sadly I don't have 25th Anniversary and haven't dumped my digital version of the game, had 2 friends test for me. Luckily it's easy to replicate but kind of time-consuming. Someone else in the discord did it too and also noticed the SELF files weren't being decrypted, but I don't know if it's related.

[RPCS3 v0.0.15-12071-5ca63816.log](https://github.com/RPCS3/rpcs3/files/9067012/RPCS3.v0.0.15-12071-5ca63816.log)
[RPCS3 v0.19-13069-0ab36ef4.log](https://github.com/RPCS3/rpcs3/files/9067015/RPCS3.v0.19-13069-0ab36ef4.log)
[RPCS3 vv0.0.18-12860-02e1c498.log](https://github.com/RPCS3/rpcs3/files/9067016/RPCS3.vv0.0.18-12860-02e1c498.log)
[RPCS3 vv0.0.23-13857-3840015e.log](https://github.com/RPCS3/rpcs3/files/9067011/RPCS3.vv0.0.23-13857-3840015e.log)

**Comment by ItsALH** [comment]
It's also in-game, most of the time when loading a new area. 

For example, in Act 2 there's a cutscene that happens when you get close to a vehicle, that one in particular won't have to compile when showing the loading screen because the area has been compiled before you got to that point.

**Comment by ItsALH** [comment]
> I had the same issue on my copy BLUS30109. I found that if I cleared all caches and then launched the game with ppu precompilation selected it would pre-load all modules for the entire game. You might try clearing profile (save games - system data) before launching as well because that setting might affect the "installation act by act" or "total installation" feature in-game. I can't remember if this process broke my save or not so be cautious. I found that launching the game even once without ppu precompilation selected would make precompilation impossible without doing the above process. The precompilation process takes a while as well, the developers put a ton of modules in this game, for me it was about 30-40 minutes, maybe more.
> 
> Basically if you want to precompile and cache the whole game's modules the caches need to be 100% clear before launching the game WITH precompilation selected. If the game is loaded without precompilation selected with the caches clear it will only load necessary modules and the game will compile load screen by load screen even if you turn the setting back on.

Did you read anything in the first post? xD

**Comment by Snakeskiin** [comment]
Was trying to help, what did I miss?

**Comment by Snakeskiin** [comment]
> > I had the same issue on my copy BLUS30109. I found that if I cleared all caches and then launched the game with ppu precompilation selected it would pre-load all modules for the entire game. You might try clearing profile (save games - system data) before launching as well because that setting might affect the "installation act by act" or "total installation" feature in-game. I can't remember if this process broke my save or not so be cautious. I found that launching the game even once without ppu precompilation selected would make precompilation impossible without doing the above process. The precompilation process takes a while as well, the developers put a ton of modules in this game, for me it was about 30-40 minutes, maybe more.
> > Basically if you want to precompile and cache the whole game's modules the caches need to be 100% clear before launching the game WITH precompilation selected. If the game is loaded without precompilation selected with the caches clear it will only load necessary modules and the game will compile load screen by load screen even if you turn the setting back on.
> 
> Did you read anything in the first post? xD

Was trying to help, what did I miss?

**Comment by ItsALH** [comment]
> Was trying to help, what did I miss?

Main post and elad's reply.

> PPU LLVM Precompilation works **correctly** in the disc version (v1.00), this means that after going through the whole process you won't compile in loading screens, but **if it's updated to v2.00** and precompiled, you'll have to compile modules in each loading screen.

It's my bad that I didn't explicitly state that the tests were done by completely compiling, opening the game to check and then removing the cache before precompiling again. But I considered it was obvious, so omitted it. 

mb if I sounded mad or something, just found the situation funny xD

**Comment by Snakeskiin** [comment]
> > Was trying to help, what did I miss?
> 
> Main post and elad's reply.
> 
> > PPU LLVM Precompilation works **correctly** in the disc version (v1.00), this means that after going through the whole process you won't compile in loading screens, but **if it's updated to v2.00** and precompiled, you'll have to compile modules in each loading screen.
> 
> It's my bad that I didn't explicitly state that the tests were done by completely compiling, opening the game to check and then removing the cache before precompiling again. But I considered it was obvious, so omitted it.
> 
> mb if I sounded mad or something, just found the situation funny xD

All good for real I'm also new and I don't wanna make anyone mad either xD. That really sucks because I had to do the same thing with my copy since I was having the same issue and it worked. The only other thing I could think of that I did was backup/delete/remove the system data and save game in save games which would force the game to open like brand new, maybe that could work? If that's useless I'm so sorry haha.

**Comment by Snakeskiin** [comment]
Sorry to hear that I'm sorry I couldn't help :/

**Comment by AniLeo** [comment]
Can still reproduce on v0.0.27-15035-73dba6d6

**Comment by elad335** [comment]
Test with #14090

**Comment by elad335** [comment]
Not fixed unfortunately, need to reverse keys in code for it to wirk.

**Comment by YoussefElbasha** [comment]
So any version of the game that is listed as version 02.00 in rpsc3 experiences this bug?

**Comment by ItsLaguna** [comment]
> So any version of the game that is listed as version 02.00 in rpsc3 experiences this bug?

No, only the versions that can be updated to 2.00, base 2.00 ones don't.
Also has nothing to do with issue you mentioned this in.

**Comment by MosesFrnchToast** [comment]
Is there a step by step guide to getting this game fully functional? I'd like to stream the game from my PC. I've gotten into the game to test functionality once and it froze on me. Spiked the CPU to 100%, which seems wild to me since I'm assuming my CPU is OP compared to the PS3 CPU (but I'm not well versed on CPUs so I guess I could be wrong?) I'm not looking to really do anything special, I just want to be able to play it how I did back in the day on PS3. Preferably with "Full Installation" & no message of "compiling shaders" or "ppu" notifications in the bottom left corner that takes me out of the experience.

Processor: AMD Ryzen 9 3900X 12-Core Processor 3.80 GHz
Ram: 32GB
GPU: NVIDIA 3060

**Comment by StrackChen** [comment]
Right-clicking the game and selecting Create LLVM Cache seems to solve the problem of compile modules in each loading screen.
